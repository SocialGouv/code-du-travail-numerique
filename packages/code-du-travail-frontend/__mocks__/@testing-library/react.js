/* eslint-disable import/export */
import { theme } from "@socialgouv/cdtn-ui";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
