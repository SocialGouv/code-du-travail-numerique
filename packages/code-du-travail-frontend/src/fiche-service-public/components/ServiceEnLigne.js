import { theme } from "@socialgouv/cdtn-ui";
import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import { getText } from "../utils";
import { LinkedTile } from "../../common/tiles/LinkedTile";

class ServiceEnLigne extends React.PureComponent {
  render() {
    const { data } = this.props;
    const type = data.attributes.type;
    const url = data.attributes.URL;
    const title = getText(data.children[0]);
    const source = getText(data.children[1]);
    return (
      <StyledTile
        wide
        href={url}
        rel="noopener noreferrer"
        target="_blank"
        aria-label={`${type} ${title} (nouvelle fenÃªtre)`}
        className="no-after"
        subtitle={type}
        title={title}
      >
        Source: {source}
      </StyledTile>
    );
  }
}

ServiceEnLigne.propTypes = {
  data: PropTypes.object.isRequired,
};

export default ServiceEnLigne;
const { spacings } = theme;

const StyledTile = styled(LinkedTile)`
  margin-bottom: ${spacings.base};
`;
