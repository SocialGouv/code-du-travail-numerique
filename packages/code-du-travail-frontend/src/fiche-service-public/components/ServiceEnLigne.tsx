import React from "react";

import { getText } from "../utils";
import { FicheSPDataElement } from "./ElementBuilder";
import Link from "next/link";
import { fr } from "@codegouvfr/react-dsfr";

export const ServiceEnLigne = ({ data }: { data: FicheSPDataElement }) => {
  const url = data.attributes.URL;
  const title = getText(data.children[0]);
  return (
    <Link
      className={fr.cx("fr-btn", "fr-my-2w")}
      href={url}
      target="_blank"
      rel="noopener noreferrer"
    >
      {title}
    </Link>
  );
};

export default ServiceEnLigne;
