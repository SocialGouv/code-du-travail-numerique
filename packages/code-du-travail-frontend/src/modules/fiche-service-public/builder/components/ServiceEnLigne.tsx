import React from "react";

import { getText } from "../utils";

import { fr } from "@codegouvfr/react-dsfr";
import { FicheSPDataServiceEnLigne } from "../type";
import { cleanUrl } from "./LienExterne";
import Link from "../../../common/Link";

export const ServiceEnLigne = ({
  data,
}: {
  data: FicheSPDataServiceEnLigne;
}) => {
  const url = cleanUrl(data.attributes.URL);
  const title = getText(data.children[0]);
  return (
    <div>
      <Link
        className={fr.cx("fr-btn", "fr-btn--secondary", "fr-mb-4w")}
        href={url}
        target="_blank"
        rel="noopener noreferrer"
      >
        {title}
      </Link>
    </div>
  );
};

export default ServiceEnLigne;
