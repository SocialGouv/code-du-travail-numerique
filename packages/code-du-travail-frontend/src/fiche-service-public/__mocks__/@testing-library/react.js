import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@socialgouv/cdtn-ui";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
