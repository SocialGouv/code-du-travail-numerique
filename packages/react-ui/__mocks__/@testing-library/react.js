import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

import { colors } from "../../src/theme";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
