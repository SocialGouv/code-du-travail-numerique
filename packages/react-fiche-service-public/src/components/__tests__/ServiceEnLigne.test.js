import { render } from "@testing-library/react";
import React from "react";

import ServiceEnLigne from "../ServiceEnLigne";
import serviceEnLigneDataMock from "./mocks/serviceEnLigneData.json";

describe("<ServiceEnLigne />", () => {
  it("should render", () => {
    const { container } = render(
      <ServiceEnLigne data={serviceEnLigneDataMock} />
    );
    expect(container).toMatchSnapshot();
  });
});
