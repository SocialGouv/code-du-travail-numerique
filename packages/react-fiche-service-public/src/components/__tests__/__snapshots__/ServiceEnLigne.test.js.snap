// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceEnLigne /> should render 1`] = `
.c4 {
  display: block;
  margin: 0 0 1rem 0;
  color: #4d73b8;
  font-weight: 600;
  font-size: 1.2rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.625;
  text-transform: uppercase;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  margin: 0;
  padding: 2rem 2rem;
  color: #3e486e;
  font-weight: normal;
  font-size: 1.6rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: none;
  border-radius: 0.6rem;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.2);
  cursor: pointer;
  -webkit-transition: box-shadow 250ms linear, -webkit-transform 100ms linear;
  -webkit-transition: box-shadow 250ms linear, transform 100ms linear;
  transition: box-shadow 250ms linear, transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:hover,
.c0:active,
.c0:focus {
  color: #3e486e;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.4);
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c2 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

.c3 {
  padding-right: 1rem;
}

.c5 {
  margin: 0;
  font-size: 1.8rem;
  font-weight: 600;
}

.c6 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-top: 1rem;
}

.c1 {
  margin-bottom: 1.6rem;
}

@media (max-width:600px) {
  .c0 {
    padding: 1.6rem 1.6rem;
    font-size: 1.4rem;
  }
}

<div>
  <a
    aria-label="Se retrouve dans en entête Texte qui se retrouve dans le lien (nouvelle fenêtre)"
    class="c0 c1 no-after"
    href="url à retoruver dans le lien"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <p
          class="c4 "
        >
          Se retrouve dans en entête
        </p>
        <p
          class="c5"
        >
          Texte qui se retrouve dans le lien
        </p>
      </div>
    </div>
    <div
      class="c6"
    >
      Source: 
      Source qui se retrouve sous le lien
    </div>
  </a>
</div>
`;
